sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachine.java

```diff
@@ -1952,6 +1959,17 @@ interface WithUserData {
             WithCreate withUserData(String base64EncodedUserData);
         }
 
+        /** The stage of a virtual machine definition allowing to specify capacity reservation configurations. */
+        interface WithCapacityReservation {
+            /**
+             * Specifies the id of capacity reservation groups for the virtual machine.
+             *
+             * @param capacityReservationGroupsId the id of capacity reservation groups
+             * @return the next stage of the definition
+             */
+            WithCreate withCapacityReservationGroupsId(String capacityReservationGroupsId);
```

Comment:
Let us be consistent. I think singular in "capacity reservation group" be more appropriate in these APIs. And getter already be singluar.

```suggestion
            WithCreate withCapacityReservationGroupId(String capacityReservationGroupId);
```

Also update the javadoc to use singular.

@XiaofeiCao Any suggestion about whether for the setter, we use `withCapacityReservationGroup` or `withCapacityReservationGroupId`?

Reply:
@XiaofeiCao @weidongxu-microsoft 
I think `withCapacityReservationGroup` would be better, and getter can also be changed to `capacityReservationGroup`.

Reply:
Personally, I think we can use `withCapacityReservationGroup(String capacityReservationGroupId)` for setter. For getter, we should use `capacityReservationGroupId`, to distinguish between a capacity reservation group instance(there's no override for return type).

Reply:
Fixed in the [commit#d3df8a2](https://github.com/Azure/azure-sdk-for-java/pull/42914/commits/d3df8a26b67b9430da2e4363e5b7c50ae8bfa316)